pkgdb/src/flox-flake.cc

```diff
@@ -37,22 +37,84 @@
 namespace flox {
 
 /* -------------------------------------------------------------------------- */
+void
+ensureFlakeIsDownloaded( std::function<void()> && lambda )
+{
+  pid_t pid = fork();
+  if ( pid == -1 )
+    {
+      // WML - TODO - better error handling here!
+      errorLog( nix::fmt(
+        "ensureFlakeIsDownloaded: faild to fork for flake downlod!" ) );
+      exit( -1 );
+    }
+  if ( 0 < pid )
+    {
+      debugLog(
+        nix::fmt( "ensureFlakeIsDownloaded: waiting for child:%d", pid ) );
+      int status = 0;
+      waitpid( pid, &status, 0 );
+      debugLog( nix::fmt(
+        "ensureFlakeIsDownloaded: child is finished, exitcode:%d, signal: %d",
+        WEXITSTATUS( status ),
+        WTERMSIG( status ) ) );
+
+      if ( WIFEXITED( status ) )
+        {
+          if ( WEXITSTATUS( status ) == EXIT_SUCCESS )
+            {
+              // The flake should be downloaded and cached locally now
+              // return to the caller.
+              return;
+            }
+          else
+            {
+              // The error has already been reported via the child, just pass
+              // along the exit code.
+              exit( WEXITSTATUS( status ) );
+            }
+        }
+      else { throw LockFlakeException( "failed to lock flake" ); }
+    }
+  else
+    {
+      lambda();
+      try
+        {
+          debugLog(
+            nix::fmt( "ensureFlakeIsDownloaded(child): finished, exiting" ) );
```

Comment:
```suggestion
            "ensureFlakeIsDownloaded(child): finished, exiting" );
```
`nix::fmt` is only necessary for `printf` style formatting.
Removing it may avoid allocations.